go-controller/pkg/ovn/base_network_controller_policy.go

```diff
@@ -1638,16 +1715,20 @@ func getStaleNetpolAddrSetDbIDs(policyNamespace, policyName, policyType, idx, co
 	})
 }
 
-func (oc *DefaultNetworkController) getNetpolDefaultACLDbIDs(direction string) *libovsdbops.DbObjectIDs {
-	return libovsdbops.NewDbObjectIDs(libovsdbops.ACLNetpolDefault, oc.controllerName,
+func (bnc *BaseNetworkController) getNetpolDefaultACLDbIDs(direction string) *libovsdbops.DbObjectIDs {
+	return libovsdbops.NewDbObjectIDs(libovsdbops.ACLNetpolDefault, bnc.controllerName,
 		map[libovsdbops.ExternalIDKey]string{
 			libovsdbops.ObjectNameKey:      allowHairpinningACLID,
 			libovsdbops.PolicyDirectionKey: direction,
 		})
 }
 
-func (oc *DefaultNetworkController) addHairpinAllowACL() error {
+func (bnc *BaseNetworkController) addHairpinAllowACL() error {
 	var v4Match, v6Match, match string
+
+	if bnc.IsSecondary() {
```

Comment:
that means we'd need another set of networkControllerPolicyEventHandler only to handle policy related events for default network ,and they are all same (copy and paste) execpt this one. I'd prefer not to do that.

Reply:
Yeah I understand, what I mean is `addHairpinAllowACL` is called from `syncNetworkPoliciesCommon`, but it may be only called from `syncNetworkPolicies` instead, because it does nothing for `syncMultiNetworkPolicies`, so there is no sense in including it to `syncNetworkPoliciesCommon`?

Reply:
ok, that needs to move retries of policy from networkControllerPolicyEventHandler to default/layer3/baseLayer3/NetworkControllerEventHandler. I've made the change.